### PR TITLE
Add test case around handling file names with spaces.

### PR DESCRIPTION
**Pull request by samccone**
This is specifically for https://github.com/sveltejs/kit/issues/740 to
assert that sirv is working as expected.

---

additional refactoring was done for the new test to DRY up the test cases by looping between dev and prod.

### CODE REVIEW

**Comment by lukeed** [comment]
Hey, thanks, but can you please revert the refactoring & only add your new test case? I'd much rather be repetitive & err on the side of clarity.

**Comment by samccone** [comment]
Done.

File: tests/sirv.js [hunk 1]
```diff
@@ -123,6 +123,32 @@ encode('should work when the request path contains encoded characters :: prod',
 	}
 });
 
+encode(`should work when the request path contains space encoded :: dev`, async () => {
+	let server = utils.http({ dev:  true});
```
Comment (lukeed) [inline tests/sirv.js:127]:
```suggestion
	let server = utils.http({ dev:  true });
```

File: tests/sirv.js [hunk 2]
```diff
@@ -123,6 +123,32 @@ encode('should work when the request path contains encoded characters :: prod',
...
+		assert.is(res.headers['content-type'], 'text/plain');
+		assert.is(res.data, 'with space.txt\n');
+		assert.is(res.statusCode, 200);
+	} finally {
+		server.close();
+	}
+});
+
+encode(`should work when the request path contains space encoded :: prod`, async () => {
+	let server = utils.http({ dev: false});
```
Comment (lukeed) [inline tests/sirv.js:140]:
```suggestion
	let server = utils.http({ dev: false });
```

**Review by lukeed — APPROVED** [review]
Thank you :)

**Pull request merged**